File: spring-web-java/src/main/java/spring/web/java/presentation/exception/EntityNotFoundException.java
package spring.web.java.presentation.exception;

import org.springframework.http.HttpStatus;

public class EntityNotFoundException extends BusinessException {

	public EntityNotFoundException(Class<?> clazz, Long id) {
		super("%s 엔티티를 찾을 수 없습니다. (ID: %s)".formatted(clazz.getSimpleName(), id), HttpStatus.NOT_FOUND);
	}
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/presentation/exception/EntityNotFoundException.kt
package spring.web.kotlin.presentation.exception

import org.springframework.http.HttpStatus

class EntityNotFoundException(clazz: Class<*>, id: Long) :
    AbstractHttpException("${clazz.simpleName} 엔티티를 찾을 수 없습니다. (ID: $id)", HttpStatus.NOT_FOUND)